fscryptctl.c

```diff
@@ -416,7 +416,7 @@ static int cmd_add_key(int argc, char *const argv[]) {
 }
 
 static int cmd_remove_key(int argc, char *const argv[]) {
-  int ioc = FS_IOC_REMOVE_ENCRYPTION_KEY;
+  unsigned long ioc = FS_IOC_REMOVE_ENCRYPTION_KEY;
```

Comment:
Great catch!  I think this was still guaranteed to work, though.  The ioctl codes do get incorrectly sign-extended when passed to the `unsigned long request` argument, but the actual Linux system call has `unsigned int cmd`.  So the sign-extended bits get ignored.

`xfs_io` has similar code too (for xfstests); I'll have to fix that too.

Reply:
The issue seems to be that `FS_IOC_REMOVE_ENCRYPTION_KEY` is a value (`0xC0406618`) that exceeds INT_MAX (`0x7FFFFFFF`) so assigning it to an `int` overflows, and signed overflow is UB. Using `unsigned long` or `unsigned int` should be fine, as they're both `uint32_t` on Linux.

Reply:
I just used `unsigned long` as that's what's used in the [ioctl manpage](https://man7.org/linux/man-pages/man2/ioctl.2.html).

Reply:
Assigning a value > INT_MAX && <= UINT_MAX to an int is actually implementation-defined behavior, not undefined behavior; the rules for integer conversions are different from the rules for integer overflow.  Anything that runs Linux will have the expected behavior here.  Anyway, the actual change is fine, I am just complaining about the explanation :-)